code/modules/industrial_lift/elevator/elevator_machinery.dm

```diff
@@ -373,3 +389,385 @@
+
+MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/button/elevator, 32)
+
+
+/**
+ * A lift indicator aka an elevator hall lantern w/ floor number
+ */
+/obj/machinery/lift_indicator
+	name = "elevator indicator"
+	desc = "Indicates what floor the elevator is at and which way it's going."
+	icon = 'icons/obj/machines/lift_indicator.dmi'
+	icon_state = "lift_indo-base"
+	base_icon_state = "lift_indo-"
+	max_integrity = 500
+	integrity_failure = 0.25
+	idle_power_usage = BASE_MACHINE_IDLE_CONSUMPTION * 0.05
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * 0.02
+	anchored = TRUE
+	density = FALSE
+
+	light_range = 1
+	light_power = 1
+	light_color = LIGHT_COLOR_DARK_BLUE
+	luminosity = 1
+
+	maptext_x = 17
+	maptext_y = 21
+	maptext_width = 4
+	maptext_height = 8
+
+	/// What specific_lift_id do we link with?
+	var/linked_elevator_id
+
+	// = (real lowest floor's z-level) - (what we want to display)
+	var/lowest_floor_offset = 1
+
+	/// Weakref to the lift.
+	var/datum/weakref/lift_ref
+	/// The lowest floor number. Determined by lift init.
+	var/lowest_floor_num = 1
+	/// Positive for going up, negative going down, 0 for stopped
+	var/current_lift_direction = 0
+	/// The lift's current floor relative to its lowest floor being 1
+	var/current_lift_floor = 1
+
+/obj/machinery/lift_indicator/Initialize(mapload)
+	. = ..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/lift_indicator/LateInitialize()
+	. = ..()
+
+	for(var/datum/lift_master/possible_match as anything in GLOB.active_lifts_by_type[BASIC_LIFT_ID])
+		if(possible_match.specific_lift_id != linked_elevator_id)
+			continue
+
+		lift_ref = WEAKREF(possible_match)
+		RegisterSignal(possible_match, COMSIG_LIFT_SET_DIRECTION, PROC_REF(on_lift_direction))
+
+/obj/machinery/lift_indicator/examine(mob/user)
+	. = ..()
+
+	if(!is_operational)
+		. += span_notice("The display is dark.")
+		return
+
+	var/dirtext
+	switch(current_lift_direction)
+		if(UP)
+			dirtext = "travelling upwards"
+		if(DOWN)
+			dirtext = "travelling downwards"
+		else
+			dirtext = "stopped"
+
+	. += span_notice("The elevator is at floor [current_lift_floor], [dirtext].")
+
+/**
+ * Update state, and only process if lift is moving.
+ */
+/obj/machinery/lift_indicator/proc/on_lift_direction(datum/source, direction)
+	SIGNAL_HANDLER
+
+	var/datum/lift_master/lift = lift_ref?.resolve()
+	if(!lift)
+		return
+
+	set_lift_state(direction, current_lift_floor)
+	update_operating()
+
+/obj/machinery/lift_indicator/on_set_is_operational()
+	. = ..()
+
+	update_operating()
+
+/**
+ * Update processing state.
+ *
+ * Returns whether we are still processing.
+ */
+/obj/machinery/lift_indicator/proc/update_operating()
+	// Let process() figure it out to have the logic in one place.
+	var/should_process = process() != PROCESS_KILL
+	if(should_process)
+		begin_processing()
+		return
+	end_processing()
+
+/obj/machinery/lift_indicator/process()
```

Comment:
this should really return some explicit value that isn't null for when we invoke it in `update_operating()`

Reply:
Would TRUE be fine?